include/nbl/video/ISwapchain.h

```diff
@@ -0,0 +1,67 @@
+#ifndef __NBL_I_SWAPCHAIN_H_INCLUDED__
+#define __NBL_I_SWAPCHAIN_H_INCLUDED__
+
+#include "nbl/core/IReferenceCounted.h"
+#include "nbl/video/surface/ISurface.h"
+#include "nbl/video/IGPUImage.h"
+#include "nbl/video/IGPUSemaphore.h"
+#include "nbl/video/IGPUFence.h"
+#include "nbl/video/IGPUImage.h"
+
+namespace nbl {
+namespace video
+{
+
+class ISwapchain : public core::IReferenceCounted
+{
+public:
+    struct SCreationParams
+    {
+        core::smart_refctd_ptr<ISurface> surface;
+        uint32_t minImageCount;
+        ISurface::SFormat surfaceFormat;
+        ISurface::E_PRESENT_MODE presentMode;
+        uint32_t width;
+        uint32_t height;
+        uint32_t arrayLayers = 1u;
+        core::smart_refctd_dynamic_array<uint32_t> queueFamilyIndices;
+
+        //VkImageUsageFlags imageUsage;
+        //VkSharingMode imageSharingMode;
+        //VkSurfaceTransformFlagBitsKHR preTransform;
+        //VkCompositeAlphaFlagBitsKHR compositeAlpha;
+        //VkBool32 clipped;
+        //VkSwapchainKHR oldSwapchain;
+    };
+
+    enum E_ACQUIRE_IMAGE_RESULT
+    {
+        EAIR_SUCCESS,
+        EAIR_TIMEOUT,
+        EAIR_NOT_READY,
+        EAIR_SUBOPTIMAL,
+        EAIR_ERROR
+    };
+
+    uint32_t getImageCount() const { return m_images->size(); }
+    core::SRange<core::smart_refctd_ptr<IGPUImage>> getImages()
+    {
+        return { m_images->begin(), m_images->end() };
+    }
+
+    virtual E_ACQUIRE_IMAGE_RESULT acquireNextImage(uint64_t timeout, IGPUSemaphore* semaphore, IGPUFence* fence, uint32_t* out_imgIx) = 0;
```

Comment:
maybe use the actual `std::chrono::duration` named type for this (is timeout in miliseconds?)